scripts/zones/Halvung/npcs/_1qp.lua

```diff
@@ -0,0 +1,34 @@
+-----------------------------------
+-- Area: Halvung
+--  NPC: Decorative Bronze Gate (_1qp)
+-----------------------------------
+local ID = require("scripts/zones/Halvung/IDs")
+require("scripts/globals/npc_util")
+require("scripts/globals/items")
+require("scripts/globals/status")
+-----------------------------------
+local entity = {}
+
+entity.onTrade = function(player, npc, trade)
+    if npcUtil.tradeHasExactly(trade, {2221, 2222, 2223}) then
+        player:confirmTrade()
+        npc:openDoor()
+        player:messageSpecial(ID.text.KEY_BREAKS, 2221,2222,2223)
+    end
+end
+
+entity.onTrigger = function(player, npc)
+    if player:getZPos() <= 79 and npc:getAnimation() == xi.anim.CLOSE_DOOR then -- from inside the door
```

Comment:
Thought not sure if it works like does the invisible door turn to face you are not but if not
If npc:isFacing(player) then or
If not npc:isFacing(player) then

If this does work could fix all them doorways to Staging points

Reply:
which invisible door(s) are you referring to?

Reply:
Don't mind me I was thinking out loud